statsmodels/base/tests/test_penalized.py

```diff
@@ -0,0 +1,728 @@
+        cls.res1 = mod.fit(cov_type=cov_type, method='bfgs', maxiter=100,
+                           disp=0)
+
+        cls.exog_index = slice(None, cls.k_nonzero, None)
+
+        cls.atol = 5e-6
+        cls.rtol = 1e-6
+
+
+class TestPenalizedGLMGaussianOracleHC2(CheckPenalizedGaussian):
+    # with trimming
+
+    @classmethod
+    def _initialize(cls):
+        y, x = cls.y, cls.x
+        # adding 10 to avoid strict rtol at predicted values close to zero
+        y = y + 10
+        cov_type = 'HC0'
+        modp = GLM(y, x[:, :cls.k_nonzero], family=family.Gaussian())
+        cls.res2 = modp.fit(cov_type=cov_type, method='bfgs', maxiter=100,
+                            disp=0)
+
+        mod = GLMPenalized(y, x, family=family.Gaussian(), penal=cls.penalty)
+        mod.pen_weight *= 1.5  # same as discrete Poisson
+        mod.penal.tau = 0.05
+        cls.res1 = mod.fit(cov_type=cov_type, method='bfgs', maxiter=100,
+                           disp=0, trim=True)
+
+        cls.exog_index = slice(None, cls.k_nonzero, None)
+        cls.k_params = cls.k_nonzero
+        cls.atol = 1e-5
+        cls.rtol = 1e-5
+
+
+class TestPenalizedGLMGaussianL2(CheckPenalizedGaussian):
+    # L2 penalty on redundant exog
+
+    @classmethod
+    def _initialize(cls):
+        y, x = cls.y, cls.x
+        # adding 10 to avoid strict rtol at predicted values close to zero
+        y = y + 10
+        cov_type = 'HC0'
+        modp = GLM(y, x[:, :cls.k_nonzero], family=family.Gaussian())
+        cls.res2 = modp.fit(cov_type=cov_type, method='bfgs', maxiter=100,
+                            disp=0)
+
+        weights = (np.arange(x.shape[1]) >= 4).astype(float)
+        mod = GLMPenalized(y, x, family=family.Gaussian(),
+                           penal=smpen.L2ContraintsPenalty(weights=weights))
+        # make pen_weight large to force redundant to close to zero
+        mod.pen_weight *= 500
+        cls.res1 = mod.fit(cov_type=cov_type, method='bfgs', maxiter=100,
+                           disp=0, trim=False)
+
+        cls.exog_index = slice(None, cls.k_nonzero, None)
+        cls.k_params = x.shape[1]
+        cls.atol = 1e-5
+        cls.rtol = 1e-5
+
+
+class TestPenalizedGLMGaussianL2Theil(CheckPenalizedGaussian):
+    # L2 penalty on redundant exog
+
+    @classmethod
+    def _initialize(cls):
+        y, x = cls.y, cls.x
+        # adding 10 to avoid strict rtol at predicted values close to zero
+        y = y + 10
+        k = x.shape[1]
+        cov_type = 'HC0'
+        restriction = np.eye(k)[2:]
+        modp = TheilGLS(y, x, r_matrix=restriction)
+        # the corresponding Theil penweight seems to be 2 * nobs / sigma2_e
+        cls.res2 = modp.fit(pen_weight=120.74564413221599 * 1000)
+
+
+        pen = smpen.L2ContraintsPenalty(restriction=restriction)
+        mod = GLMPenalized(y, x, family=family.Gaussian(),
+                           penal=pen)
+        # make pen_weight large to force redundant to close to zero
```

Comment:
Does multiplying by 1 (next line) accomplish the goal described in this comment?